### PR TITLE
Add axis labels, zero value and legend to PageDashboardChart

### PR DESCRIPTION
**Pull request by ZitaNemeckova**
TODO: 
PageDashboardChart changes:
- [x] Add axis labels
- [x] Add legend
- [x] Address bug that does not allow for zero values
- [x] Address bug that causes y-axis ticks to mess up when values are zero 

**Add axis labels:**

Example:
```diff
--- a/frontend/awx/dashboard/charts/JobsChart.tsx
+++ b/frontend/awx/dashboard/charts/JobsChart.tsx
@@ -73,6 +73,7 @@ export function JobsChart(props: {
         { color: errorColor, values: error },
         { color: canceledColor, values: canceled },
       ]}
+      settings={{ minDomain: 0, xLabel: 'x', yLabel: 'y' }}
     />
   );
 }
```
Screenshot 2023-07-20 at 11 45 58

**Zero values are allowed:**

Example:
```diff
diff --git a/frontend/awx/dashboard/charts/JobsChart.tsx b/frontend/awx/dashboard/charts/JobsChart.tsx
index 97647c96..009396c5 100644
--- a/frontend/awx/dashboard/charts/JobsChart.tsx
+++ b/frontend/awx/dashboard/charts/JobsChart.tsx
@@ -73,6 +73,7 @@ export function JobsChart(props: {
         { color: errorColor, values: error },
         { color: canceledColor, values: canceled },
       ]}
+      settings={{ minDomain: 0, xLabel: 'x label', yLabel: 'y label', allowZero: true }}
     />
   );
 }
```
Screenshot 2023-07-20 at 11 47 47

**Add legend:**

Example:
```diff
diff --git a/frontend/awx/dashboard/charts/JobsChart.tsx b/frontend/awx/dashboard/charts/JobsChart.tsx
index 97647c96..7c217eb8 100644
--- a/frontend/awx/dashboard/charts/JobsChart.tsx
+++ b/frontend/awx/dashboard/charts/JobsChart.tsx
@@ -73,6 +73,12 @@ export function JobsChart(props: {
         { color: errorColor, values: error },
         { color: canceledColor, values: canceled },
       ]}
+      settings={{
+        xLabel: 'x label',
+        yLabel: 'y label',
+        allowZero: true,
+        legendData: [{ name: 'legend' }, { name: 'john' }],
+      }}
     />
   );
 }
```
Screenshot 2023-07-20 at 12 30 08

### CODE REVIEW

File: framework/PageDashboard/PageDashboardChart.tsx [hunk 1]
```diff
@@ -16,24 +18,66 @@ export function PageDashboardChart(props: {
...
+    // label for x ax
+    yLabel?: string;
+    // label for y ax
+    xLabel?: string;
+    // prevents filtering of zero values
+    allowZero?: boolean;
+    // chart will have lines instead of area
+    useLines?: boolean;
+    // data to be shown in the legend
+    legendData?: {
```
Comment (jamestalton) [inline framework/PageDashboard/PageDashboardChart.tsx:35]: The legend should be based off the groups that are passed in by the developer.
That makes it so the dev does not have to keep the legend and groups in sync externally to this component.
SEE: https://github.com/ansible/ansible-ui/pull/781/files#diff-4b17c85c4a9d09cfb0e23bbcc5655cae1662a45515c215ca029119e5fd1d3d0cL43

File: framework/PageDashboard/PageDashboardChart.tsx [hunk 2]
```diff
@@ -16,24 +18,66 @@ export function PageDashboardChart(props: {
...
         <Chart
-          padding={{ bottom: 60, left: 60, right: 40, top: 16 }}
+          padding={{
+            bottom: settings?.xLabel ? 70 : 60,
+            left: settings?.yLabel ? 70 : 60,
+            right: 40,
+            top: 16,
+          }}
           colorScale={groups.map((group) => group.color)}
+          height={settings && settings.legendData ? size.height * 0.9 : size.height}
```
Comment (jamestalton) [inline framework/PageDashboard/PageDashboardChart.tsx:67]: If the height is tweaked here, the SVG will not render at the height of the container that contains the chart.
This causes the chart fonts to not be the correct size. This should be handled using padding.
In addition we should not be using a percentage on the height. (i.e. 0.9)
SEE: https://github.com/ansible/ansible-ui/pull/781/files#diff-4b17c85c4a9d09cfb0e23bbcc5655cae1662a45515c215ca029119e5fd1d3d0cR59

File: framework/PageDashboard/PageDashboardChart.tsx [hunk 3]
```diff
@@ -47,16 +91,38 @@ export function PageDashboardChart(props: {
...
-            {groups.map((group, index) => (
-              <ChartArea
-                key={index}
-                data={group.values.map((value) => ({ x: value.label, y: value.value }))}
-                interpolation="monotoneX"
-              />
-            ))}
+            {groups.map((group, index) =>
+              settings && settings.useLines ? (
+                <ChartLine
```
Comment (jamestalton) [inline framework/PageDashboard/PageDashboardChart.tsx:112]: Putting chart lines in a ChartStack makes the lines stack and not overlap.
The lines need to be outside of the ChartStack.
SEE:  https://github.com/ansible/ansible-ui/pull/781/files#diff-4b17c85c4a9d09cfb0e23bbcc5655cae1662a45515c215ca029119e5fd1d3d0cR87

**Review by jamestalton — CHANGES_REQUESTED** [review]

File: framework/PageDashboard/PageDashboardChart.tsx (hunk 3 above)
Comment (ZitaNemeckova) [reply framework/PageDashboard/PageDashboardChart.tsx]: Fixed.

File: framework/PageDashboard/PageDashboardChart.tsx (hunk 1 above)
Comment (ZitaNemeckova) [reply framework/PageDashboard/PageDashboardChart.tsx]: Fixed.

File: framework/PageDashboard/PageDashboardChart.tsx [hunk 4]
```diff
@@ -43,21 +85,39 @@ export function PageDashboardChart(props: {
...
-            fixLabelOverlap
-            // tickFormat={(date: string) => `${new Date(date).toLocaleDateString()}`}
-            //  tickFormat={(n) => `${Math.round(n)}`}
+            dependentAxis
+            showGrid
+            label={yLabel}
+            tickFormat={
+              onlyIntegerTicks
+                ? //eslint-disable-next-line @typescript-eslint/no-unsafe-return
+                  (t) => (Number.isInteger(t) ? t : '')
```
Comment (skochay330) [inline framework/PageDashboard/PageDashboardChart.tsx:96]: I am still seeing no tick values on the y-axis when `onlyIntegerTicks` is true- maybe try pushing the changes again?

**Review by skochay330 — CHANGES_REQUESTED** [review]
Aside from two small tweaks, looks great!

File: framework/PageDashboard/PageDashboardChart.tsx (hunk 4 above)
Comment (ZitaNemeckova) [reply framework/PageDashboard/PageDashboardChart.tsx:101]: I forgot to push the commit... Sorry. It should be there now with missing props you mentioned in your review.

**Review by skochay330 — APPROVED** [review]

**Pull request merged**